### PR TITLE
Environment setting support.

### PR DESCRIPTION
**Pull request by tasdomas**
Added new config directive 'environment', that lists environment variables to be set on the tmux session. Needs tests and documentation, but feedback if the general approach seems sensible would be very useful.

### CODE REVIEW

File: tmuxp/session.py [hunk 1]
```diff
@@ -330,6 +330,27 @@ def set_option(self, option, value):
...
+
+        proc = self.cmd(
+            'set-environment', name, value
+        )
+
+        if proc.stderr:
+            if isinstance(proc.stderr, list) and len(proc.stderr) == int(1):
+                proc.stderr = proc.stderr[0]
+            raise ValueError('tmux set-environment stderr: %s' % proc.stderr)
+
```
Comment (tony) [inline tmuxp/session.py]: thanks for the documentation

File: tmuxp/testsuite/workspacebuilder.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ def test_focus_pane_index(self):
         self.assertNotEqual(w.get('window_name'), 'man')
 
         pane_path = '/usr'
-        for i in range(10):
+        for i in range(20):
```
Comment (tony) [inline tmuxp/testsuite/workspacebuilder.py]: Fine

**Comment by tony** [comment]
Good call on this @tasdomas 

I opened a question at https://github.com/tmux/tmux/issues/304 to double check if set-environment with ``-g`` is just server only.  If so we may want to chuck a ``set_environment`` to the server as well.

Tests / documentation / examples for this would be a good idea. It seems like a feature that'd be helpful in many situations.

**Comment by tasdomas** [comment]
@tony, cool - I'll update the pull request and submit it for your consideration.

**Comment by tony** [comment]
We probably should also add ``set-environment`` at the server level.  ``-g`` / "global" is the same as ``server``.

Any updates on this @tasdomas? If you're having trouble / time constraints let me know.

**Comment by tasdomas** [comment]
@tony, working on it - currently trying to figure out a way to test the functionality. Once that's done, everything else is relatively easy. Should have a PR by next week - a little busy at the moment.

**Comment by tony** [comment]
No problem, if you need help or need me to finish the test part, I'm happy to.

**Comment by tasdomas** [comment]
Hello, @tony, I've updated the pull request. Extended the functionality to support setting global variables on the server as well, added some docs and enabled environment variable expansion for the variables (so setting something like PATH would be less tedious). Looking forward to your reviews.

File: tmuxp/testsuite/workspacebuilder.py (hunk 2 above)
Comment (tasdomas) [inline tmuxp/testsuite/workspacebuilder.py]: This test fails for me at a noticeable frequency with only 10 retries.

**Comment by tony** [comment]
@tasdomas: Pardon the delay. Merged.

This looks good to me and is useful enough of a feature to bump the version up.

Uploaded to pypi as 0.11.0.

**Pull request merged**